gateway/upstreamer/push/movingaverage.go

```diff
@@ -0,0 +1,43 @@
+package push
+
+// MovingAverage represent a moving average
+// give an number of samples.
+type MovingAverage struct {
+	samples int
+	ring    []float64
+	nextIdx int
+}
+
+// Average return the average of the samples
+// If samples are not compplete it returns 0
+func (m *MovingAverage) Average() float64 {
+
+	var sum = float64(0)
+
+	for _, value := range m.ring {
+		// we need a complete sample
+		if value == 0 {
+			return 0
```

Comment:
This should be a constant, and like a negative value. How do I differentiate 0 from not ready?